src/inference/src/dev/iplugin.cpp

```diff
@@ -400,8 +400,33 @@ std::unordered_set<std::string> ov::get_supported_nodes(
                 update_supported = false;
                 for (auto& op : model->get_ordered_ops()) {
                     const auto& name = op->get_friendly_name();
-                    if (fused_model_op_map.find(name) != fused_model_op_map.end() && supported.count(name)) {
-                        if (!supported.count(fused_model_op_map[name]) &&
+                    if (removed_nodes.count(name) && supported.count(name)) {
+                        if (has_all_consumers_unsupported(supported, op)) {
+                            supported.erase(name);
+                            removed_nodes.erase(name);
+                            update_supported = true;
+                        }
+                    }
+                }
+            }
+            // For example, A op need to be removed from supported:
+            //              A (fused on B, to be marked as unsupported)
+            //              |
+            //              B (unsupported)
+            //
+            //            A ShapeOf (to be marked as unsupported)
+            //              |
+            //              B (unsupported)
+            //
+            update_supported = true;
+            while (update_supported) {
+                update_supported = false;
+                for (auto& op : model->get_ordered_ops()) {
+                    const auto& name = op->get_friendly_name();
+                    bool is_shapeof = ov::is_type<op::util::ShapeOfBase>(op);
+                    if (((fused_model_op_map.find(name) != fused_model_op_map.end()) || is_shapeof) &&
+                        supported.count(name)) {
+                        if ((!supported.count(fused_model_op_map[name]) || is_shapeof) &&
```

Comment:
seems this is a special case(shape of) of codes line 401 to 408? 
maybe we can consider optimize in future?

Reply:
It is a problem that occurs when recursively looping through the entire graph, move shape of case to L401-L408 can not fix this issues, we will try the simpler way to optimize this API

Reply:
Please create a ticket to follow up  if there is a way to make this solution more common.